libs/base/utilities/src/misc/intersect-arrays.ts

```diff
@@ -19,17 +19,32 @@ export function intersectArrays<T>(
     arrays = [arr1OrIdentityFn, ...arrays];
   }
 
-  const intersectingMap = new Map<unknown, T>();
+  const intersectionArray: T[] = [];
+  const [firstArray, ...restArrays] = arrays;
 
-  for (const array of arrays) {
-    for (const item of array) {
-      const id = identityFn(item);
+  firstArray.filter((value) => {
```

Comment:
This implementation seems like a mess, there are functional methods used but not as expected, filter should be filtering items but it's not, map should be mapping items but it's not.
Let's fix this function to properly use functional style:
1. Use filter to actually filter items of `firstArray` and assign the result to `intersectionArray`
2. Replace `restArrays.map` with `restArrays.every` to check that in every other array the item is found
3. Replace `array.map` with `array.some` to check that item exists in specific array

All of the mutable let variables are no longer needed as the state is managed by the functional array methods which you can now use. So on high level the pipeline should be like this:
```ts
firstArray.filter -> restArrays.every -> array.some
```

Reply:
Updated